mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1629,6 +1552,327 @@ func TestClientSideEncryptionProse(t *testing.T) {
+			{
+				opts := options.DataKey().SetKeyAltNames([]string{defKeyAltName})
+
+				var err error
+				defKeyID, err = cse.clientEnc.CreateDataKey(context.Background(), "local", opts)
+				assert.Nil(mt, err, "error creating %q data key: %v", defKeyAltName, err)
+			}
+			return defKeyID
+		}
+
+		mt.Run("case 1: createKey()", func(mt *mtest.T) {
+			initialize()
+
+			// Use client_encryption to create a new local data key with a keyAltName "abc" and assert the operation
+			// does not fail.
+			{
+				opts := options.DataKey().SetKeyAltNames([]string{abcKeyAltName})
+				_, err := cse.clientEnc.CreateDataKey(context.Background(), "local", opts)
+				assert.Nil(mt, err, "error creating %q data key: %v", abcKeyAltName, err)
+			}
+
+			// Repeat Step 1 and assert the operation fails due to a duplicate key server error (error code 11000).
+			{
+				opts := options.DataKey().SetKeyAltNames([]string{abcKeyAltName})
+				_, err := cse.clientEnc.CreateDataKey(context.Background(), "local", opts)
+				assert.NotNil(mt, err, "duplicate %q key did not propagate expected error", abcKeyAltName)
+
+				e110000 := "E11000 duplicate key"
+				correctError := strings.Contains(err.Error(), e110000)
+				assert.True(t, correctError, "expected error to contain %q, got: %v", e110000, err)
+			}
+
+			// Use client_encryption to create a new local data key with a keyAltName "def" and assert the operation
+			// fails due to a duplicate key server error (error code 11000).
+			{
+				opts := options.DataKey().SetKeyAltNames([]string{abcKeyAltName})
+				_, err := cse.clientEnc.CreateDataKey(context.Background(), "local", opts)
+				assert.NotNil(mt, err, "duplicate %q key did not propagate expected error", defKeyAltName)
+
+				e110000 := "E11000 duplicate key"
+				correctError := strings.Contains(err.Error(), e110000)
+				assert.True(t, correctError, "expected error to contain %q, got: %v", e110000, err)
+			}
+		})
+
+		mt.Run("case 2: addKeyAltName()", func(t *mtest.T) {
+			defKeyID := initialize()
+
+			var someNewKeyID primitive.Binary
+			// Use client_encryption to create a new local data key and assert the operation does not fail.
+			{
+				var err error
+				someNewKeyID, err = cse.clientEnc.CreateDataKey(context.Background(), "local")
+				assert.Nil(mt, err, "error creating data key: %v", err)
+			}
+
+			// Use client_encryption to add a keyAltName "abc" to the key created in Step 1 and assert the operation
+			// does not fail.
+			{
+				res := cse.clientEnc.AddKeyAltName(context.Background(), someNewKeyID, abcKeyAltName)
+				assert.Nil(mt, res.Err(), "error adding key alt name: %v", res.Err())
+			}
+
+			// Repeat Step 2, assert the operation does not fail, and assert the returned key document contains the
+			// keyAltName "abc" added in Step 2.
+			{
+				res := cse.clientEnc.AddKeyAltName(context.Background(), someNewKeyID, abcKeyAltName)
+				assert.Nil(mt, res.Err(), "error adding key alt name: %v", res.Err())
```

Comment:
Also check that the returned key document contains the keyAltName "abc".

Reply:
The [unified test for adding key alt names ](https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/unified/addKeyAltName.yml#L159) expect that we don't return the updated result for `AddKeyAltName`.  But for completeness, I will add a function that does a read to ensure the data has been set correctly.